filesystem_service/src/filesystem_service/monitor.py

```diff
@@ -46,92 +48,49 @@ def unregister_subscriber(self, subs):
 
         self._subscribers.remove(subs)
 
-    @property
-    def exclude_folder(self):
-        """
-        :return: list of folders to be excluded from monitoring
-        """
-
-        return self._exclude_folder
-
     def add_exclude_folder(self, folder):
         """
         :param folder:full path to folder. Type: string
         :return:
         """
 
-        self._exclude_folder.add(folder.encode())
-
-    @property
-    def folder(self):
-        """
-        :return: Monitored folder
-        """
-
-        return self._folder
-
-    @folder.setter
-    def folder(self, folder):
-        """
-        :param folder: full path to folder to be monitored. Type: string
-        :return:
-        """
-
-        self._folder = folder.encode()
-
-    def start(self):
-        """
-        Start process that generates filesystem events
-        :return:
-        """
-
-        self._process = Process(target=self.notify)
-        self._process.start()
-
-    def stop(self):
-        """
-        Terminate filesystem event generator process
-        :return:
-        """
-
-        if self._process:
-            self._process.terminate()
+        self._exclude_folders.add(folder.encode())
 
-    def gen_watch_list(self):
+    def __gen_watch_list(self):
         """
         :return: list of directoried to watch without excludes
         """
 
-        return [res[0]for res in os.walk(self.folder)
-                if res[0] not in self.exclude_folder]
+        return [res[0] for res in os.walk(self._root_folder)
+                if res[0] not in self._exclude_folders]
 
-    def add_watch(self, folder):
+    def __add_watch(self, folder):
         """
         :param folder: full path to folder to be watched. Type: string
         :return:
         """
 
-        self.notifier.add_watch(folder.encode())
+        self._notifier.add_watch(folder.encode())
 
-    def remove_watch(self, folder):
+    def __remove_watch(self, folder):
         """
         :param folder: full path to folder to be removed from watch list.
         Type: string
         :return:
         """
 
-        self.notifier.remove_watch(folder.encode())
+        self._notifier.remove_watch(folder.encode())
 
-    def notify(self):
+    def monitor(self):
         """
         Notify subscribers about filesystem events
         :return:
         """
 
-        for folder in self.gen_watch_list():
-            self.notifier.add_watch(folder)
+        for folder in self.__gen_watch_list():
+            self._notifier.add_watch(folder)
 
-        for event in self.notifier.event_gen():
+        for event in self._notifier.event_gen():
             if event is not None:
```

Comment:
if event:

Reply:
Modified. flake is silent like a grave. BTW, that's a piece of code from documentation.

Reply:
@freishutz, comparison with `None` is `x == None`.